File: solune/backend/src/models/activity.py
"""Pydantic models for the activity log / audit trail."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel


class ActivityEvent(BaseModel):
    """Response model for a single activity event."""

    id: str
    event_type: str
    entity_type: str
    entity_id: str
    project_id: str
    actor: str
    action: str
    summary: str
    detail: dict[str, Any] | None = None
    created_at: str


class ActivityEventCreate(BaseModel):
    """Internal model used by log_event() — no id/created_at (DB-generated)."""

    event_type: str
    entity_type: str
    entity_id: str
    project_id: str
    actor: str = "system"
    action: str
    summary: str
    detail: dict[str, Any] | None = None


class ActivityStats(BaseModel):
    """Aggregated activity statistics for a project."""

    total_count: int
    today_count: int
    by_type: dict[str, int]
    last_event_at: str | None = None

File: solune/frontend/src/types/activity.ts
/**
 * Activity event types (054-activity-audit-trail).
 */

export interface ActivityEvent {
  id: string;
  event_type: string;
  entity_type: string;
  entity_id: string;
  project_id: string;
  actor: string;
  action: string;
  summary: string;
  detail?: Record<string, unknown>;
  created_at: string;
}

export interface ActivityStats {
  total_count: number;
  today_count: number;
  by_type: Record<string, number>;
  last_event_at: string | null;
}
